src/noit_message_decoder.c

```diff
@@ -660,8 +701,8 @@ noit_metric_tagset_init_with_context(noit_metric_tagset_t *lookup,
     free((void *) tags);
     return -1;
   }
-  lookup->tags = realloc(tags, tag_count * sizeof(noit_metric_tag_t));
-  lookup->tag_count = tag_count;
+  lookup->tags = realloc(tags, parsed_tag_count * sizeof(noit_metric_tag_t));
```

Comment:
Prior to this change, this realloc was pointless because it would always realloc the same size that was previously calloced.
But, even with this change, what good is a realloc here if the tags have already been set by `noit_metric_tags_parse_with_context`? 
When would `parsed_tag_count` ever be different than `tag_count`? 
If `parsed_tag_count` > `tag_count`, `noit_metric_tags_parse_with_context` will have already written beyond the originally allocated memory. if that's the case, then this realloc has the potential to lose that memory if it can't extend the allocation and needs to copy elsewhere. `noit_metric_tags_parse_with_context` also runs the risk of overwriting other memory since it would be writing past the previously calloced memory.

Reply:
don't know, would need to pretty much take over this ticket to figure that out

Reply:
this realloc is correct in it's current (`parsed_tag_count`) state.

`noit_metric_tags_count` returns the number of commas + 1 as the `tag_count`.
`parsed_tag_count` could end up being returned less than `tag_count` if tags do not match the validation function or if there are duplicate tags.
This means that the realloc will never allocate __more__ memory. It can only ever reduce the allocated memory.